hibernate-core/src/main/java/org/hibernate/type/descriptor/java/Immutability.java

```diff
@@ -23,10 +23,10 @@ public class Immutability implements MutabilityPlan<Object> {
 	/**
 	 * Singleton access
 	 */
-	public static final Immutability INSTANCE = new Immutability();
+	private static final Immutability INSTANCE = new Immutability();
```

Comment:
Please revert and deprecate for removal instead. This is an SPI, so we'd have to first deprecate the field.